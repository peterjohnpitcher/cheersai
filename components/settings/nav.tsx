'use client'

import Link from 'next/link'
import { usePathname } from 'next/navigation'
import {
  User,
  Palette,
  Image,
  Lock,
  CreditCard,
  Share2,
  Calendar,
  Bell,
  Users,
  ChevronLeft
} from 'lucide-react'

const settingsNavItems = [
  {
    name: 'Account',
    href: '/settings/account',
    icon: User,
    description: 'Personal information and preferences'
  },
  {
    name: 'Brand',
    href: '/settings/brand',
    icon: Palette,
    description: 'Brand voice and identity'
  },
  {
    name: 'Logo & Watermark',
    href: '/settings/logo',
    icon: Image,
    description: 'Logos and watermark settings'
  },
  {
    name: 'Team',
    href: '/settings/team',
    icon: Users,
    description: 'Team members and permissions'
  },
  {
    name: 'Security',
    href: '/settings/security',
    icon: Lock,
    description: 'Password and security settings'
  },
  {
    name: 'Billing',
    href: '/settings/billing',
    icon: CreditCard,
    description: 'Subscription and payment'
  },
  {
    name: 'Social Connections',
    href: '/settings/connections',
    icon: Share2,
    description: 'Connected social media accounts'
  },
  {
    name: 'Posting Schedule',
    href: '/settings/posting-schedule',
    icon: Calendar,
    description: 'Default posting times'
  },
  {
    name: 'Notifications',
    href: '/settings/notifications',
    icon: Bell,
    description: 'Email and notification preferences'
  }
]

export function SettingsNav() {
  const pathname = usePathname()
  
  return (
    <nav className="w-full md:w-64 flex-shrink-0">
      <div className="bg-white rounded-large shadow-sm border border-border p-2">
        <Link
          href="/dashboard"
          className="flex items-center gap-2 px-3 py-2 text-sm text-text-secondary hover:text-text-primary transition-colors mb-2"
        >
          <ChevronLeft className="w-4 h-4" />
          Back to Dashboard
        </Link>
        
        <div className="border-t border-border my-2" />
        
        <div className="space-y-1">
          {settingsNavItems.map((item) => {
            const Icon = item.icon
            const isActive = pathname === item.href
            
            return (
              <Link
                key={item.href}
                href={item.href}
                className={`
                  flex items-start gap-3 px-3 py-2.5 rounded-medium transition-colors
                  ${isActive 
                    ? 'bg-primary/10 text-primary' 
                    : 'hover:bg-gray-50 text-text-secondary hover:text-text-primary'
                  }
                `}
              >
                <Icon className="w-5 h-5 mt-0.5 flex-shrink-0" />
                <div className="flex-1 min-w-0">
                  <div className={`text-sm font-medium ${isActive ? 'text-primary' : ''}`}>
                    {item.name}
                  </div>
                  <div className="text-xs text-text-secondary mt-0.5 hidden lg:block">
                    {item.description}
                  </div>
                </div>
              </Link>
            )
          })}
        </div>
      </div>
    </nav>
  )
}