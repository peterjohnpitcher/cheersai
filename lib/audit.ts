import { createClient } from '@/lib/supabase/server'

export async function auditLog(opts: { tenantId?: string; userId?: string; entityType: string; entityId: string; action: string; meta?: Record<string, unknown> }) {
  const supabase = await createClient()
  await supabase.from('audit_log').insert({
    tenant_id: opts.tenantId || null,
    user_id: opts.userId || null,
    entity_type: opts.entityType,
    entity_id: opts.entityId,
    action: opts.action,
    meta: opts.meta || null,
  })
}

export async function addPostRevision(opts: { postId: string; userId?: string; diff: Record<string, unknown>; version?: number }) {
  const supabase = await createClient()
  // Determine next version
  const { data: rows } = await supabase
    .from('post_revisions')
    .select('version')
    .eq('post_id', opts.postId)
    .order('version', { ascending: false })
    .limit(1)
  const nextVersion = (rows?.[0]?.version || 0) + 1
  await supabase.from('post_revisions').insert({
    post_id: opts.postId,
    user_id: opts.userId || null,
    version: opts.version ?? nextVersion,
    diff: opts.diff as any,
  })
}

