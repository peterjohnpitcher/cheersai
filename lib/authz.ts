import { createClient } from '@/lib/supabase/server'

export async function hasPermission(userId: string, tenantId: string, permission: string): Promise<boolean> {
  const supabase = await createClient()
  const { data, error } = await supabase
    .from('user_roles')
    .select('role_id, roles:roles!inner(name, tenant_id, id), perms:role_permissions!inner(permission)')
    .eq('tenant_id', tenantId)
    .eq('user_id', userId)
  if (error) return false
  const rows = data || []
  if (rows.length === 0) {
    // Backward-compat: if no roles configured, allow operation
    return true
  }
  return rows.some((r: any) => r.perms?.permission === permission)
}

export const PERMISSIONS = {
  POST_CREATE: 'post.create',
  POST_APPROVE: 'post.approve',
  POST_PUBLISH: 'post.publish',
  CONNECTIONS_MANAGE: 'connections.manage',
  BILLING_MANAGE: 'billing.manage',
  ROLES_MANAGE: 'roles.manage',
} as const
