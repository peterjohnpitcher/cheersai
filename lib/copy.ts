export const TERMS = {
  GBP: 'Google Business Profile',
  PLAN_PRO: 'Professional',
  PLAN_STARTER: 'Starter',
  PLAN_ENTERPRISE: 'Enterprise',
} as const;

export function formatPlanLabel(tier?: string | null): string | null {
  if (!tier) return null;
  const t = tier.toLowerCase();
  if (t === 'pro' || t === 'professional') return TERMS.PLAN_PRO;
  if (t === 'starter' || t === 'free') return TERMS.PLAN_STARTER;
  if (t === 'enterprise' || t === 'business') return TERMS.PLAN_ENTERPRISE;
  return tier.charAt(0).toUpperCase() + tier.slice(1);
}

