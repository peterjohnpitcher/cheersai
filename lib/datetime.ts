export function getUserTimeZone(): string {
  try {
    // In browsers, prefer the userâ€™s resolved IANA timezone
    if (typeof Intl !== 'undefined' && Intl.DateTimeFormat) {
      const tz = Intl.DateTimeFormat().resolvedOptions().timeZone
      if (tz) return tz
    }
  } catch {}
  // Fallback to London if unavailable
  return 'Europe/London'
}

type DateInput = Date | string | number

function toDate(value: DateInput): Date {
  return value instanceof Date ? value : new Date(value)
}

export function formatTime(value: DateInput, tz?: string): string {
  const date = toDate(value)
  const timeZone = tz || getUserTimeZone()
  // en-GB with explicit 12-hour and lowercase am/pm
  return new Intl.DateTimeFormat('en-GB', {
    hour: 'numeric',
    minute: '2-digit',
    hour12: true,
    timeZone,
  }).format(date).toLowerCase()
}

export function formatDate(value: DateInput, tz?: string, opts?: Intl.DateTimeFormatOptions): string {
  const date = toDate(value)
  const timeZone = tz || getUserTimeZone()
  const options: Intl.DateTimeFormatOptions = opts ?? { year: 'numeric', month: 'short', day: '2-digit' }
  return new Intl.DateTimeFormat('en-GB', { ...options, timeZone }).format(date)
}

export function formatDateTime(value: DateInput, tz?: string, opts?: Intl.DateTimeFormatOptions): string {
  const date = toDate(value)
  const timeZone = tz || getUserTimeZone()
  const defaults: Intl.DateTimeFormatOptions = {
    year: 'numeric', month: 'short', day: '2-digit',
    hour: 'numeric', minute: '2-digit', hour12: true,
  }
  const options = { ...(opts || {}), timeZone }
  return new Intl.DateTimeFormat('en-GB', { ...defaults, ...options }).format(date).replace(/AM|PM/g, m => m.toLowerCase())
}

