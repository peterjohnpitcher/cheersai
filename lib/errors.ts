// Centralized error codes and mapping helpers

export type ErrorCode =
  | 'TOKEN_EXPIRED'
  | 'MISSING_SCOPE'
  | 'IG_IMAGE_REQUIRED'
  | 'RATE_LIMITED'
  | 'FORBIDDEN'
  | 'UNAUTHORIZED'
  | 'NOT_FOUND'
  | 'INVALID_INPUT'
  | 'PROVIDER_ERROR'
  | 'NETWORK_ERROR'
  | 'SERVER_ERROR'
  | 'UNKNOWN';

export type Provider = 'facebook' | 'instagram' | 'twitter' | 'gbp' | 'generic';

export interface AppError {
  code: ErrorCode;
  message: string;
  details?: unknown;
}

function includes(msg: string, ...needles: string[]) {
  const lower = msg.toLowerCase();
  return needles.some(n => lower.includes(n));
}

export function mapProviderError(err: unknown, provider: Provider = 'generic'): AppError {
  const raw = normalizeError(err);
  const msg = raw.message || '';
  const status = raw.statusCode || raw.status || 0;

  // HTTP status mapping
  if (status === 401) return { code: 'UNAUTHORIZED', message: scrubMessage(msg, provider) };
  if (status === 403) return { code: 'FORBIDDEN', message: scrubMessage(msg, provider) };
  if (status === 404) return { code: 'NOT_FOUND', message: scrubMessage(msg, provider) };
  if (status === 429) return { code: 'RATE_LIMITED', message: 'Rate limited. Please retry later.' };

  // Provider-specific patterns
  if (provider === 'instagram' || provider === 'facebook') {
    if (includes(msg, 'expired token', 'token expired', 'session has expired')) {
      return { code: 'TOKEN_EXPIRED', message: 'Please reconnect your account.' };
    }
    if (includes(msg, 'permissions error', 'missing permission', 'missing scope', 'insufficient permission')) {
      return { code: 'MISSING_SCOPE', message: 'Missing required permission for this action.' };
    }
    if (provider === 'instagram' && includes(msg, 'requires an image', 'image required', 'media_url is required')) {
      return { code: 'IG_IMAGE_REQUIRED', message: 'Instagram posts require at least one image.' };
    }
  }

  if (provider === 'twitter') {
    if (includes(msg, 'rate limit', 'too many requests', '429')) {
      return { code: 'RATE_LIMITED', message: 'Twitter rate limit hit. Please retry later.' };
    }
    if (includes(msg, 'token', 'invalid', 'expired')) {
      return { code: 'TOKEN_EXPIRED', message: 'Please reconnect your Twitter account.' };
    }
  }

  if (includes(msg, 'network', 'fetch failed', 'timeout', 'timed out')) {
    return { code: 'NETWORK_ERROR', message: 'Network error. Please try again.' };
  }

  // Generic fallbacks
  if (status >= 500) return { code: 'SERVER_ERROR', message: 'Upstream service error. Please retry.' };
  if (status >= 400) return { code: 'PROVIDER_ERROR', message: scrubMessage(msg, provider) };

  return { code: 'UNKNOWN', message: scrubMessage(msg, provider) };
}

export function normalizeError(err: unknown): { message: string; status?: number; statusCode?: number; name?: string; raw?: unknown } {
  if (!err) return { message: 'Unknown error' };
  if (typeof err === 'string') return { message: err };
  if (err instanceof Error) return { message: err.message, name: err.name, raw: err } as any;
  try {
    const anyErr: any = err as any;
    const msg = anyErr.message || JSON.stringify(anyErr);
    return { message: msg, status: anyErr.status, statusCode: anyErr.statusCode, raw: err };
  } catch {
    return { message: 'Unknown error' };
  }
}

// Ensure we do not bubble provider internals directly
function scrubMessage(message: string, provider: Provider): string {
  const m = message?.trim();
  if (!m) return 'Request failed';
  // Remove long JSON bodies and tokens
  return m.replace(/[A-Za-z0-9_\-]{16,}/g, '[redacted]')
          .replace(/"access_token"\s*:\s*"[^"]+"/g, '"access_token":"[redacted]"')
          .slice(0, 500);
}

