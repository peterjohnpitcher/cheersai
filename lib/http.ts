import { NextRequest, NextResponse } from 'next/server'

export type RequestContext = { requestId: string }
export type ApiOk<T> = { ok: true; data: T; requestId: string }
export type ApiErr = { ok: false; error: { code: string; message: string; details?: unknown }; requestId: string }
export type ApiResult<T> = ApiOk<T> | ApiErr

function requestIdFromHeaders(req?: NextRequest): string | undefined {
  return req?.headers.get('x-request-id') || undefined
}

function generateRequestId(): string {
  if (typeof globalThis.crypto?.randomUUID === 'function') {
    return globalThis.crypto.randomUUID()
  }
  return `${Date.now()}-${Math.random().toString(16).slice(2)}`
}

export function getRequestContext(req?: NextRequest): RequestContext {
  return { requestId: requestIdFromHeaders(req) || generateRequestId() }
}

export function ok<T>(data: T, reqOrCtx?: NextRequest | Partial<RequestContext>, init?: ResponseInit) {
  const ctx = isNextRequest(reqOrCtx) ? getRequestContext(reqOrCtx) : mergeCtx(reqOrCtx)
  return NextResponse.json<ApiOk<T>>({ ok: true, data, requestId: ctx.requestId }, { status: 200, ...init })
}

export function badRequest(code: string, message: string, details?: unknown, reqOrCtx?: NextRequest | Partial<RequestContext>) {
  const ctx = isNextRequest(reqOrCtx) ? getRequestContext(reqOrCtx) : mergeCtx(reqOrCtx)
  return NextResponse.json<ApiErr>({ ok: false, error: { code, message, details }, requestId: ctx.requestId }, { status: 400 })
}

export function unauthorized(message = 'Unauthorized', details?: unknown, reqOrCtx?: NextRequest | Partial<RequestContext>) {
  const ctx = isNextRequest(reqOrCtx) ? getRequestContext(reqOrCtx) : mergeCtx(reqOrCtx)
  return NextResponse.json<ApiErr>({ ok: false, error: { code: 'unauthorized', message, details }, requestId: ctx.requestId }, { status: 401 })
}

export function forbidden(message = 'Forbidden', details?: unknown, reqOrCtx?: NextRequest | Partial<RequestContext>) {
  const ctx = isNextRequest(reqOrCtx) ? getRequestContext(reqOrCtx) : mergeCtx(reqOrCtx)
  return NextResponse.json<ApiErr>({ ok: false, error: { code: 'forbidden', message, details }, requestId: ctx.requestId }, { status: 403 })
}

export function notFound(message = 'Not Found', details?: unknown, reqOrCtx?: NextRequest | Partial<RequestContext>) {
  const ctx = isNextRequest(reqOrCtx) ? getRequestContext(reqOrCtx) : mergeCtx(reqOrCtx)
  return NextResponse.json<ApiErr>({ ok: false, error: { code: 'not_found', message, details }, requestId: ctx.requestId }, { status: 404 })
}

export function serverError(message = 'Internal server error', details?: unknown, reqOrCtx?: NextRequest | Partial<RequestContext>) {
  const ctx = isNextRequest(reqOrCtx) ? getRequestContext(reqOrCtx) : mergeCtx(reqOrCtx)
  return NextResponse.json<ApiErr>({ ok: false, error: { code: 'server_error', message, details }, requestId: ctx.requestId }, { status: 500 })
}

export function rateLimited(message = 'Too many requests', retryAfterSeconds?: number, details?: unknown, reqOrCtx?: NextRequest | Partial<RequestContext>) {
  const ctx = isNextRequest(reqOrCtx) ? getRequestContext(reqOrCtx) : mergeCtx(reqOrCtx)
  const headers: Record<string, string> = {}
  if (typeof retryAfterSeconds === 'number' && retryAfterSeconds > 0) {
    headers['Retry-After'] = String(Math.ceil(retryAfterSeconds))
  }
  return NextResponse.json<ApiErr>({ ok: false, error: { code: 'RATE_LIMITED', message, details }, requestId: ctx.requestId }, { status: 429, headers })
}

function isNextRequest(input: unknown): input is NextRequest {
  return typeof input === 'object' && input !== null && 'headers' in input &&
    typeof (input as NextRequest).headers?.get === 'function'
}

function mergeCtx(ctx?: Partial<RequestContext>): RequestContext {
  return { requestId: ctx?.requestId || generateRequestId() }
}
