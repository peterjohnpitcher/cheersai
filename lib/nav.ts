export type IconName =
  | 'Send'
  | 'Plus'
  | 'Image'
  | 'Calendar'
  | 'BarChart3'
  | 'Activity'
  | 'Settings'
  | 'Home'
  | 'FileText'
  | 'Clock'
  | 'Palette'
  | 'Link2'
  | 'Mic'
  | 'MapPin'
  | 'Users'
  | 'CreditCard'
  | 'Shield'
  | 'Bell';

export type NavItem = {
  id?: string;
  label: string;
  to: string; // Relative or absolute path
  icon?: IconName;
  badge?: string | number;
  requiresPlan?: 'starter' | 'professional' | 'enterprise';
  requiresRole?: 'admin' | 'editor' | 'viewer';
  requiresConnection?: boolean;
};

export const mainNav: NavItem[] = [
  { id: 'dashboard', label: 'Dashboard', to: '/dashboard' },
  { id: 'campaigns', label: 'Campaigns', to: '/campaigns' },
  { id: 'media', label: 'Media', to: '/media' },
  { id: 'settings', label: 'Settings', to: '/settings' },
];

export type SubNavPreset =
  | 'dashboard'
  | 'campaignsRoot'
  | 'campaignDetail'
  | 'settings'
  | 'media'
  | 'publishing'
  | 'notifications'
  | 'admin';

export const subNavPresets: Record<SubNavPreset, NavItem[]> = {
  dashboard: [
    { label: 'Quick Post', to: '#quick-post', icon: 'Send' },
    { label: 'New Campaign', to: '/campaigns/new', icon: 'Plus' },
    { label: 'Media', to: '/media', icon: 'Image' },
    { label: 'Campaigns', to: '/campaigns', icon: 'Calendar' },
    { label: 'Settings', to: '/settings', icon: 'Settings' },
  ],
  campaignsRoot: [
    // Canonical section actions for Campaigns
    { label: 'New Campaign', to: 'new', icon: 'Plus' },
    { label: 'Queue Monitor', to: '/publishing/queue', icon: 'Clock' },
  ],
  campaignDetail: [
    { label: 'Publishing', to: 'publishing', icon: 'Send', requiresConnection: true },
  ],
  settings: [
    { label: 'Account', to: '', icon: 'Settings' },
    { label: 'Brand', to: 'brand', icon: 'Palette' },
    { label: 'Guardrails', to: 'guardrails', icon: 'Shield' },
    { label: 'Logo & Watermark', to: 'logo', icon: 'Image' },
    { label: 'Team', to: 'team', icon: 'Users' },
    { label: 'Security', to: 'security', icon: 'Shield' },
    { label: 'Billing', to: 'billing', icon: 'CreditCard' },
    { label: 'Connections', to: 'connections', icon: 'Link2' },
    { label: 'Posting Schedule', to: 'posting-schedule', icon: 'Clock' },
    { label: 'Notifications', to: 'notifications', icon: 'Bell' },
  ],
  media: [
    { label: 'Library', to: '', icon: 'Image' },
    { label: 'New Campaign', to: '/campaigns/new', icon: 'Plus' },
  ],
  publishing: [
    { label: 'Queue Monitor', to: 'queue', icon: 'Clock' },
    { label: 'New Campaign', to: '/campaigns/new', icon: 'Plus' },
  ],
  notifications: [
    { label: 'Notifications', to: '', icon: 'Bell' },
    { label: 'Notification Settings', to: '/settings/notifications', icon: 'Bell' },
  ],
  admin: [
    { label: 'Dashboard', to: 'dashboard', icon: 'Home' },
    { label: 'Tenants', to: 'tenants', icon: 'Users' },
    { label: 'Content Settings', to: 'content-settings', icon: 'Settings' },
    { label: 'AI Prompts', to: 'ai-prompts', icon: 'Mic' },
  ],
  approvals: [
    { label: 'Review Queue', to: '', icon: 'Bell' },
    { label: 'Queue Monitor', to: '/publishing/queue', icon: 'Clock' },
  ],
  reports: [
    { label: 'Attribution', to: 'attribution', icon: 'BarChart3' },
    { label: 'New Campaign', to: '/campaigns/new', icon: 'Plus' },
  ],
};

export function filterNavItems(
  items: NavItem[],
  context: { plan?: string; role?: string; hasConnections?: boolean }
): NavItem[] {
  return items.filter((item) => {
    if (item.requiresPlan) {
      const planHierarchy = ['starter', 'professional', 'enterprise'];
      const requiredIndex = planHierarchy.indexOf(item.requiresPlan);
      const userIndex = planHierarchy.indexOf(context.plan || 'starter');
      if (userIndex < requiredIndex) return false;
    }
    if (item.requiresRole) {
      const roleHierarchy = ['viewer', 'editor', 'admin'];
      const requiredIndex = roleHierarchy.indexOf(item.requiresRole);
      const userIndex = roleHierarchy.indexOf(context.role || 'viewer');
      if (userIndex < requiredIndex) return false;
    }
    if (item.requiresConnection && !context.hasConnections) {
      return false;
    }
    return true;
  });
}

export function getGreetingForTimezone(timezone: string = 'Europe/London'): string {
  const now = new Date();
  const formatter = new Intl.DateTimeFormat('en-GB', { hour: 'numeric', hour12: false, timeZone: timezone });
  const hour = parseInt(formatter.format(now));
  if (hour < 12) return 'Good morning';
  if (hour < 18) return 'Good afternoon';
  return 'Good evening';
}
