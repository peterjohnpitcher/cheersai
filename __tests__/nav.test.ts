import { filterNavItems, subNavPresets, type NavItem } from '@/lib/nav';

describe('nav filtering', () => {
  test('allows all dashboard items for default context', () => {
    const items = subNavPresets.dashboard as NavItem[];
    const filtered = filterNavItems(items, {});
    expect(filtered).toMatchSnapshot();
  });

  test('respects plan requirements', () => {
    const items: NavItem[] = [
      { label: 'Starter', to: '/s', requiresPlan: 'starter' },
      { label: 'Pro', to: '/p', requiresPlan: 'professional' },
      { label: 'Enterprise', to: '/e', requiresPlan: 'enterprise' },
    ];
    expect(filterNavItems(items, { plan: 'starter' })).toMatchSnapshot('starter');
    expect(filterNavItems(items, { plan: 'professional' })).toMatchSnapshot('professional');
    expect(filterNavItems(items, { plan: 'enterprise' })).toMatchSnapshot('enterprise');
  });

  test('respects role requirements', () => {
    const items: NavItem[] = [
      { label: 'ViewerOnly', to: '/v', requiresRole: 'viewer' },
      { label: 'Editor', to: '/e', requiresRole: 'editor' },
      { label: 'Admin', to: '/a', requiresRole: 'admin' },
    ];
    expect(filterNavItems(items, { role: 'viewer' })).toMatchSnapshot('viewer');
    expect(filterNavItems(items, { role: 'editor' })).toMatchSnapshot('editor');
    expect(filterNavItems(items, { role: 'admin' })).toMatchSnapshot('admin');
  });
});

