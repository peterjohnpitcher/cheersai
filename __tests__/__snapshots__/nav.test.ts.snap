// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nav filtering allows all dashboard items for default context 1`] = `
[
  {
    "icon": "Send",
    "label": "Quick Post",
    "to": "#quick-post",
  },
  {
    "icon": "Plus",
    "label": "New Campaign",
    "to": "/campaigns/new",
  },
  {
    "icon": "Image",
    "label": "Media",
    "to": "/media",
  },
  {
    "icon": "Calendar",
    "label": "Campaigns",
    "to": "/campaigns",
  },
  {
    "icon": "Settings",
    "label": "Settings",
    "to": "/settings",
  },
]
`;

exports[`nav filtering respects plan requirements: enterprise 1`] = `
[
  {
    "label": "Starter",
    "requiresPlan": "starter",
    "to": "/s",
  },
  {
    "label": "Pro",
    "requiresPlan": "professional",
    "to": "/p",
  },
  {
    "label": "Enterprise",
    "requiresPlan": "enterprise",
    "to": "/e",
  },
]
`;

exports[`nav filtering respects plan requirements: professional 1`] = `
[
  {
    "label": "Starter",
    "requiresPlan": "starter",
    "to": "/s",
  },
  {
    "label": "Pro",
    "requiresPlan": "professional",
    "to": "/p",
  },
]
`;

exports[`nav filtering respects plan requirements: starter 1`] = `
[
  {
    "label": "Starter",
    "requiresPlan": "starter",
    "to": "/s",
  },
]
`;

exports[`nav filtering respects role requirements: admin 1`] = `
[
  {
    "label": "ViewerOnly",
    "requiresRole": "viewer",
    "to": "/v",
  },
  {
    "label": "Editor",
    "requiresRole": "editor",
    "to": "/e",
  },
  {
    "label": "Admin",
    "requiresRole": "admin",
    "to": "/a",
  },
]
`;

exports[`nav filtering respects role requirements: editor 1`] = `
[
  {
    "label": "ViewerOnly",
    "requiresRole": "viewer",
    "to": "/v",
  },
  {
    "label": "Editor",
    "requiresRole": "editor",
    "to": "/e",
  },
]
`;

exports[`nav filtering respects role requirements: viewer 1`] = `
[
  {
    "label": "ViewerOnly",
    "requiresRole": "viewer",
    "to": "/v",
  },
]
`;
